sqrl-planner/src/main/java/com/datasqrl/engine/database/DatabasePhysicalPlan.java

```diff
@@ -4,13 +4,23 @@
 package com.datasqrl.engine.database;
 
 import com.datasqrl.engine.EnginePhysicalPlan;
+import com.datasqrl.engine.database.relational.ddl.statements.CreateIndexDDL;
+import com.datasqrl.engine.database.relational.ddl.statements.DropIndexDDL;
 import com.datasqrl.plan.queries.APIQuery;
 
 import com.datasqrl.plan.queries.IdentifiedQuery;
+import com.datasqrl.sql.SqlDDLStatement;
+import java.util.List;
 import java.util.Map;
 
 public interface DatabasePhysicalPlan extends EnginePhysicalPlan {
 
+  List<SqlDDLStatement> getDdl();
+
+  default boolean removeIndexDdl() {
+    return getDdl().removeIf(ddl -> (ddl instanceof CreateIndexDDL) || (ddl instanceof DropIndexDDL));
```

Comment:
i think the root issue is that the engine also delefates to create 'normal' base jdbc tables. It is unnecessary in duckdb as no schema needs to be created, so no DDL indices should be created at all.

Reply:
Yes, this is a bit of a hack. IMHO the root is that "indexes" don't exist for table formats and associated query engines (you can only partition data but not index yet), yet indexes are important for normal databases, which is the foundation of the JDBCEngine.

I think we should have a "BaseJDBCEngine" which only does table creation and then extend it to a OperationalJDBCEngine (i.e. postgres, mysql, sqlserver, etc) that supports indexes and subscribe/notify and AnalyticJDBCEngine which does not. That would be a cleaner seperation.